src/tests/Microsoft.PowerFx.Core.Tests/TexlTests.cs

```diff
@@ -3620,6 +3620,54 @@ public void TestAttachmentDottedNameNodeBinding(string script, string expectedSc
             symbol.AddEntity(new TestDataSource("DS", schema));
 
             TestSimpleBindingSuccess(script, expectedType, symbol);
+        }
+
+        [Theory]
+        [InlineData("Search(T, \"a\", \"Name\")", "*[Name:s]")]
+        [InlineData("Search(T, If(1<0,\"a\"), \"Name\")", "*[Name:s]")]
+        [InlineData("Search(T2,\"a\", \"Name\", \"Address\")", "*[Name:s,Age:n,Address:s]")]
+        public void TexlFunctionTypeSemanticsSearch(string script, string expectedType)
+        {
+            var symbol = new SymbolTable();
+            symbol.AddVariable("T", new TableType(TestUtils.DT("*[Name:s]")));
+            symbol.AddVariable("T2", new TableType(TestUtils.DT("*[Name:s,Age:n,Address:s]")));
+
+            TestSimpleBindingSuccess(
+                script,
+                TestUtils.DT(expectedType),
+                symbol,
+                new Features { ConsistentOneColumnTableResult = true });
```

Comment:
>ConsistentOneColumnTableResult = true [](http://example.com/codeflow?start=31&length=37)

There's no need to use this feature as we don't have any functions in the tests that depend on it.